### DISCUSSION TITLE
Event PlotSetForSaleEvent

### DISCUSSION THREAD

**Discussion by Fe1son** [TownyAdvanced/Towny#6129] [Q&A] [answered]
How to cancel PlotSetForSaleEvent?

**Comment by ItsJuls** [comment]
It extends event right? So i assume event.setCancelled()?

**Reply by Fe1son** [reply]
there are not such method in this event

**Comment by LlmDl** [comment, marked as answer]
PlotSetForSaleEvent doesn't implement Cancellable. You could use the event to get the townblock, wait 1 tick, set the townblock not for sale and save the townblock.

**Reply by Fe1son** [reply]
Can you please provide a code example of how this is done?

**Reply by LlmDl** [reply]
Inside your event listener:
```java

Bukkit.getScheduler.runTaskLater(this.plugin, ()-> setTownBlockNotForSale(event.getTownBlock()), 20l);

```

A method outside of the listener method:
```java
private void setTownBlockNotForSale(TownBlock tb) {
   tb.setPrice(-1);
   tb.save();
}
```
